tests/std/tests/P2445R1_forward_like/test.compile.pass.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+
+struct owner_type {
+    inline static constinit int imut = 0;
+    static constexpr int iconst      = 0;
+
+    int m_mutobj    = 0;
+    int& m_mutlref  = imut;
+    int&& m_mutrref = move(imut);
+
+    const int m_cobj    = 0;
+    const int& m_clref  = iconst;
+    const int&& m_crref = move(iconst);
+};
+
+owner_type owner{};
+constexpr owner_type& owner_lref  = owner;
+constexpr owner_type&& owner_rref = move(owner);
+
+constexpr owner_type owner_c{};
+constexpr const owner_type& owner_clref  = owner_c;
+constexpr const owner_type&& owner_crref = move(owner_c);
+
+static_assert(is_same_v<decltype(forward_like<decltype(owner)>(owner.m_mutobj)), int&&>);

```

Comment:
I'm not fond of the `decltype` usage in these `static_assert`s, I have a mild preference to see the type spelled out. With blocks of six I admit it makes little difference, so let's so what other reviewers have to say.

Reply:
I've also avoided nested `decltype` now. I don't know whether this is wanted.